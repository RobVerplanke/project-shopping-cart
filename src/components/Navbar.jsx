import { Link } from 'react-router-dom';
import { useData } from '../context/DataContext';
import { Badge } from '@mui/material';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
import '../styles/NavBar.css';

function Navbar() {
  const { cartQuantityCounter } = useData();
  return (
    <nav aria-label="Main navigation">
      <ul className="navbar-items">
        <li>
          <Link to="home" aria-label="Go to Home Page">
            HOME
          </Link>
        </li>
        <li>
          <Link to="shop" aria-label="Go to Shop Page">
            SHOP
          </Link>
        </li>
      </ul>

      {/* Cart button is on the other side of the navigation bar */}
      <ul className="navbar-cart">
        <li>
          <Link to="cart" aria-label="Go to Shopping Cart">
            <Badge badgeContent={cartQuantityCounter} color="info">
              <ShoppingCartIcon fontSize="large" />
            </Badge>
          </Link>
        </li>
      </ul>
    </nav>
  );
}

export default Navbar;
