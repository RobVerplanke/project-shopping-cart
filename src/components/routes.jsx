import App from '../App';
import Home from './pages/HomePage';
import Shop from './pages/ShopPage';
import Cart from './pages/CartPage';
import ProductDetail from './pages/ProductDetailPage';
import Errorpage from './pages/ErrorPage';

const routes = [
  {
    path: '/',
    element: <App />,
    errorElement: <Errorpage />,
    children: [
      {
        index: true,
        element: <Home />,
      },
      {
        path: 'home',
        element: <Home />,
      },
      {
        path: 'cart',
        element: <Cart />,
      },
      {
        path: 'shop',
        element: <Shop />,
      },
      {
        path: 'shop/product/:id',
        element: <ProductDetail />,
      },
    ],
  },
];

export default routes;
