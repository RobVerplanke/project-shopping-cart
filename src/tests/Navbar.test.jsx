import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, useRoutes } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Navbar from '../components/Navbar';
import routes from '../components/routes';

function AppWithRoutes() {
  return useRoutes(routes);
}

describe('Navbar component', () => {
  it('renders navigation links correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();

    // The links to Home, Shop and Cart are rendered
    expect(screen.getByText(/Home/i)).toBeInTheDocument();
    expect(screen.getByText(/Shop/i)).toBeInTheDocument();
    expect(screen.getByText(/Cart/i)).toBeInTheDocument();
  });

  it('Home component renders initially', async () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <AppWithRoutes />
      </MemoryRouter>
    );

    // Home page is rendered
    expect(screen.getByText(/Home page/i)).toBeInTheDocument();
  });

  it('Loads Shop component after click on the Shop button', async () => {
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/']}>
        <AppWithRoutes />
      </MemoryRouter>
    );

    // Home page is initially rendered
    expect(screen.getByText(/Home page/i)).toBeInTheDocument();

    // Simulate a click on the Shop button
    await user.click(screen.getByText(/Shop/i));

    // Shop page is rendered
    expect(screen.getByText(/Shop page/i)).toBeInTheDocument();
  });

  it('Loads Cart component after click on the Cart button', async () => {
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/']}>
        <AppWithRoutes />
      </MemoryRouter>
    );

    // Home page is initially rendered
    expect(screen.getByText(/Home page/i)).toBeInTheDocument();

    // Simulate a click on the Cart button
    await user.click(screen.getByText(/Cart/i));

    // Cart page is rendered
    expect(screen.getByText(/Cart page/i)).toBeInTheDocument();
  });

  // it('Loads error page on invalid route', async () => {
  //   render(
  //     <MemoryRouter initialEntries={['/invalid-route']}>
  //       <AppWithRoutes />
  //     </MemoryRouter>
  //   );

  //   await waitFor(() =>
  //     expect(screen.getByText(/Error page/i)).toBeInTheDocument()
  //   );
  // });
});
