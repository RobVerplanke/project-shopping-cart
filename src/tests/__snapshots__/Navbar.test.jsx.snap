// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar component > renders navigation links correctly 1`] = `
<DocumentFragment>
  <nav>
    <ul>
      <li>
        <a
          href="/home"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/shop"
        >
          Shop
        </a>
      </li>
      <li>
        <a
          href="/cart"
        >
          Cart
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
